File: Self_Solving_Algorithm/Cube_Algorithms/Yperm.py
from ..Cube_Turning.Cube_Turning_CCW import CubeArrayFCCW, CubeArrayRCCW, CubeArrayUCCW
from ..Cube_Turning.Cube_Turning_CW import CubeArrayFCW, CubeArrayRCW
from ..Cube_Algorithms.Common_Moves.Sexy_Move import SexyMoveAlgorithm
from ..Cube_Algorithms.Common_Moves.Sledge_Hammer import SledgeHammerAlgorithm
import copy

YpermNotation = [
    CubeArrayFCW,
    CubeArrayRCW,
    CubeArrayUCCW,
    CubeArrayRCCW,
    CubeArrayUCCW,
    SledgeHammerAlgorithm,
    SexyMoveAlgorithm,
    CubeArrayRCCW,
    CubeArrayFCW,
    CubeArrayRCW,
    CubeArrayFCCW
]

def YpermAlgorithm(cube):
    current_state = copy.deepcopy(cube)
    for move in YpermNotation:
        current_state = move(current_state)
    return current_state

File: Algorithm/cubeAlgorithms/Yperm.js
// F R U' R' U' (R U R' F') (R U R' U') R' F R F'

import { CubeArrayFCCW, CubeArrayRCCW, CubeArrayUCCW } from "../cubeTurningCCW";
import { CubeArrayFCW, CubeArrayRCW } from "../cubeTurningCW";
import { SexyMoveAlgorithm } from "./commonMoves/sexyMove";
import { SledgeHammerAlgorithm } from "./commonMoves/sledgeHammer";

var YpermNotation = [
  CubeArrayFCW,
  CubeArrayRCW,
  CubeArrayUCCW,
  CubeArrayRCCW,
  CubeArrayUCCW,
  SledgeHammerAlgorithm,
  SexyMoveAlgorithm,
  CubeArrayRCCW,
  CubeArrayFCW,
  CubeArrayRCW,
  CubeArrayFCCW,
];

export function YpermAlgorithm(cube) {
  let CurrentState = JSON.parse(JSON.stringify(cube));
  for (let i = 0; i < YpermNotation.length; i++) {
    CurrentState = YpermNotation[i](CurrentState);
  }

  return CurrentState;
}
